lighty-core/lighty-controller/src/main/java/io/lighty/core/controller/impl/LightyControllerImpl.java

```diff
@@ -394,6 +440,79 @@ private void createConcurrentDOMDataBroker() {
         this.concurrentDOMDataBrokerOld = new LegacyDOMDataBrokerAdapter(this.concurrentDOMDataBroker);
     }
 
+    /**
+     * The first member of the cluster (leader) will create his shards. Other joining members will query
+     * the leader for snapshots of the shards.
+     */
+    private void askForShards() {
```

Comment:
all of above would make this live in K8S-specific subclass, making maintenance easier (I think)